src/app/clusters/fixed-label-server/fixed-label-server.cpp

```diff
@@ -53,18 +54,30 @@ class FixedLabelAttrAccess : public AttributeAccessInterface
 CHIP_ERROR FixedLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValueEncoder & aEncoder)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
-    DeviceLayer::AttributeList<app::Clusters::FixedLabel::Structs::LabelStruct::Type, DeviceLayer::kMaxFixedLabels> labelList;
 
-    if (DeviceLayer::PlatformMgr().GetFixedLabelList(endpoint, labelList) == CHIP_NO_ERROR)
+    DeviceLayer::DeviceInfoProvider::FixedLabelIterator * it = DeviceLayer::GetDeviceInfoProvider()->IterateFixedLabel(endpoint);
+
+    if (it)
     {
-        err = aEncoder.EncodeList([&labelList](const auto & encoder) -> CHIP_ERROR {
-            for (auto label : labelList)
-            {
-                ReturnErrorOnFailure(encoder.Encode(label));
-            }
-
-            return CHIP_NO_ERROR;
-        });
+        if (it->Count() > 0)
```

Comment:
Why do we care?  We can just `EncodeList` and if there are no items in the iterator we won't encode any list items....  Should be less code and easier to read.

Reply:
We won't call encode at all within EncodeList lamda if there is no items in the iterator, is this equivalent to EncodeEmptyList which call Encode(DataModel::List<uint8_t>()) ?

If so, we don't need this check

Reply:
> is this equivalent to EncodeEmptyList which call `Encode(DataModel::List<uint8_t>())` ?

Yes.  EncodeList will encode an empty list to represent the "erase the list" op, then call the callback and encode one IB for every list item.  If there are none, it just encodes an empty list.